Use iris flower dataset from sklearn library and use cross_val_score against following models to measure the performance of each. In the end figure out the model with best performance,

Logistic Regression
SVM
Decision Tree
Random Forest

In [80]:
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np

In [81]:
iris = load_iris()

Logistic Regression

In [82]:
lr_scores = cross_val_score(LogisticRegression(), iris.data, iris.target)
lr_scores

c:\Users\Fahmid Bin Mosaraf\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\linear_model\_logistic.py:469: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

In [83]:
np.average(lr_scores)

np.float64(0.9733333333333334)

Decision Tree

In [84]:
dt_scores = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
dt_scores

array([0.96666667, 0.96666667, 0.9       , 1.        , 1.        ])

In [85]:
np.average(dt_scores)

np.float64(0.9666666666666668)

Support Vector Machine (SVM)

In [86]:
svm_scores = cross_val_score(SVC(), iris.data, iris.target)
svm_scores

array([0.96666667, 0.96666667, 0.96666667, 0.93333333, 1.        ])

In [87]:
np.average(svm_scores)

np.float64(0.9666666666666666)

Random Forest

In [88]:
rf_scores = cross_val_score(RandomForestClassifier(), iris.data, iris.target)
rf_scores

array([0.96666667, 0.96666667, 0.93333333, 0.96666667, 1.        ])

In [89]:
np.average(rf_scores)

np.float64(0.9666666666666668)

Best score so far is from Logistic Regression: 0.9733333333333334